plugin-maven/src/main/java/com/diffplug/spotless/maven/generic/Indent.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2016 DiffPlug
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.diffplug.spotless.maven.generic;
+
+import org.apache.maven.plugins.annotations.Parameter;
+
+import com.diffplug.spotless.FormatterStep;
+import com.diffplug.spotless.generic.IndentStep;
+import com.diffplug.spotless.maven.FormatterStepConfig;
+import com.diffplug.spotless.maven.FormatterStepFactory;
+
+public class Indent implements FormatterStepFactory {
+
+	private static final int DEFAULT_NUM_SPACES_PER_TAB = 4;
+
+	@Parameter
+	private boolean withTabs;
+
+	@Parameter
+	private boolean withSpaces;
+
+	@Parameter
+	private Integer spacesPerTab;
+
+	@Override
+	public FormatterStep newFormatterStep(FormatterStepConfig config) {
+		System.out.println("withTabs: " + withTabs);
+		System.out.println("withSpaces: " + withSpaces);
+		System.out.println("spacesPerTab: " + spacesPerTab);
+
+		int amountOfSpacesPerTab = DEFAULT_NUM_SPACES_PER_TAB;
+		if (spacesPerTab != null) {
+			amountOfSpacesPerTab = spacesPerTab;
+		}
+
+		if (withSpaces && withTabs) {
+			throw new IllegalArgumentException("Must specify exactly one of 'withSpaces' or 'withTabs'.");
+		}
+
+		if (!withTabs && !withSpaces) {
+			return IndentStep.create(IndentStep.Type.SPACE, amountOfSpacesPerTab);
+		} else {
```

Comment:
I wouldn't be forgiving about this.  Easier for users to debug something that gives them a clear error message than silently guessing defaults that aren't obvious.  One way to check that the user set exactly one to true is XOR:

`if (withTabs ^ withSpaces) { createStep } else { error }`

Reply:
I thought this would match the maven convetion over configuration pattern. But I'm ok with not having defaults, if this is what spotless does for gradle.

But default includes for java, scala, etc. should be kept right?

Reply:
IMO there's two good reasons for convention over configuration:

A) The default behavior is appropriate for 80%+ of use cases.  e.g. we default to GIT_ATTRIBUTES for handling line endings
B) There are *lots* of arbitrary decisions to make, and we are providing a *comprehensive* solution.  e.g. maven's layout of source directories, or `google-java-format` formatting all kinds of things with no configuration

The problem with `indent` is that the tabs vs spaces debate is alive and well - no 80+% default.  And it's not comprehensive - there's only 2 decisions to make (tabs vs space, tab:space ratio) - so we're not really reducing the user's burden by making default decisions for the user.

Reply:
Let's keep default includes - those are appropriate for 80%+ of use cases.

Reply:
I agree to your explanation.

Reply:
fixed